process/block/preprocess/transactions.go

```diff
@@ -685,6 +685,12 @@ func (txs *transactions) computeOrderedTxs(
 		return nil, nil, process.ErrEmptyTxDataPool
 	}
 
+	sortedTransactionsProvider, ok := txShardPool.(SortedTransactionsProvider)
```

Comment:
i still do not like this. It would be better to implement the sorting function on the top of the shardedcache and use that in transactions.go always. without the need of type assertion.

Reply:
Refactored, now we have an extra component which adapts the previous cache to a `SortedTransactionsProvider`.